src/Schema/TypeRegistry.php

```diff
@@ -337,24 +337,27 @@ function () use ($objectDefinition): array {
      *
      * @param  \GraphQL\Language\AST\ObjectTypeDefinitionNode|\GraphQL\Language\AST\InterfaceTypeDefinitionNode  $typeDefinition
      *
-     * @return \Closure(): array<string, array<string, mixed>>
+     * @return \Closure(): array<string, Closure(): array<string, mixed>>
      */
     protected function makeFieldsLoader($typeDefinition): Closure
     {
         return
             /**
-             * @return array<string, array<string, mixed>>
+             * @return array<string, Closure(): array<string, mixed>>
              */
-            function () use ($typeDefinition): array {
+            static function () use ($typeDefinition): array {
                 $typeValue = new TypeValue($typeDefinition);
                 $fields = [];
 
                 foreach ($typeDefinition->fields as $fieldDefinition) {
-                    /** @var \Nuwave\Lighthouse\Schema\Factories\FieldFactory $fieldFactory */
-                    $fieldFactory = app(FieldFactory::class);
-                    $fieldValue = new FieldValue($typeValue, $fieldDefinition);
-
-                    $fields[$fieldDefinition->name->value] = $fieldFactory->handle($fieldValue);
+                    $fields[$fieldDefinition->name->value] = static function () use ($typeValue, $fieldDefinition): array {
+                        /** @var \Nuwave\Lighthouse\Schema\Factories\FieldFactory $fieldFactory */
+                        $fieldFactory = app(FieldFactory::class);
```

Comment:
Perhaps we can micro-optimize by instantiating this only once and cache it in a property of `TypeRegistry`?

Reply:
I tried that before... but you mentioned this broke your app... but this would be another pretty big improvement if that would be possible (the container is not as fast as you would think)!

But because of last time I did not dare to do this (see #961) 😅

Reply:
Since then the `FieldFactory` has actually changed to be safe to reuse for multiple fields - perhaps I was subconsciously thinking about making this optimization at some point, but then forgot to go through with it. Let's give it another go.

Reply:
In that case 💪 let me put that in!